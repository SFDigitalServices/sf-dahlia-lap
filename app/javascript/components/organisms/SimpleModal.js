import React from 'react'
import classNames from 'classnames'

import Modal from './Modal'

const SimpleModalBody = ({
  type,
  header,
  isOpen,
  primary,
  secondary,
  alert,
  invert,
  handleClose,
  onPrimaryClick,
  onSecondaryClick,
  children
}) => {
  const primaryButtonClassName = classNames({
    button: true,
    primary: !type || type === 'status',
    'alert-fill': type === 'alert'
  })

  return (
    <Modal.Body hidden={isOpen} handleClose={handleClose}>
      <Modal.Header title={header} />
      {alert && <Modal.Alert {...alert} />}
      <Modal.Content>{children}</Modal.Content>
      <Modal.Footer>
        <div className='modal-button_item modal-button_primary'>
          <button className={primaryButtonClassName} onClick={onPrimaryClick}>
            {primary}
          </button>
        </div>
        <div className='modal-button_item modal-button_secondary'>
          <button className='button no-border' onClick={onSecondaryClick}>
            {secondary}
          </button>
        </div>
      </Modal.Footer>
    </Modal.Body>
  )
}

const SimpleModal = ({ children, isOpen, handleClose, ...SimpleModalBodyProps }) => {
  return (
    <Modal isOpen={isOpen} handleClose={handleClose}>
      <SimpleModalBody handleClose={handleClose} {...SimpleModalBodyProps}>
        {children}
      </SimpleModalBody>
    </Modal>
  )
}

export default SimpleModal
