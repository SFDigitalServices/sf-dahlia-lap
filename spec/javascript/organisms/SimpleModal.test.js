import React from 'react'
import renderer from 'react-test-renderer'
import SimpleModal from 'components/organisms/SimpleModal'

describe('SimpleModal', () => {
  const onCloseClick = () => {}
  const onPrimaryClick = () => {}
  const onSecondaryClick = () => {}

  test('it should render status type successfully', () => {
    const isOpen = true
    const wrapper = mount(
        <SimpleModal
          header='Update Status'
          primary='update'
          secondary='cancel'
          isOpen={isOpen}
          onCloseClick={onCloseClick}
          onPrimaryClick={onPrimaryClick}
          onSecondaryClick={onSecondaryClick}
          type='status'>
          <div>content</div>
        </SimpleModal>,
    )

    expect(wrapper).toMatchSnapshot();
  })
})
