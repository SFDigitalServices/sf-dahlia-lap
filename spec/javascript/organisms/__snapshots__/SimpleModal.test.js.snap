// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimpleModal it should render status type successfully 1`] = `
<SimpleModal
  header="Update Status"
  isOpen={true}
  onCloseClick={[Function]}
  onPrimaryClick={[Function]}
  onSecondaryClick={[Function]}
  primary="update"
  secondary="cancel"
  type="status"
>
  <Modal
    isOpen={true}
  >
    <Modal
      ariaHideApp={true}
      bodyOpenClassName="ReactModal__Body--open"
      closeTimeoutMS={0}
      isOpen={true}
      parentSelector={[Function]}
      portalClassName="ReactModalPortal"
      shouldCloseOnEsc={true}
      shouldCloseOnOverlayClick={true}
      shouldFocusAfterRender={true}
      shouldReturnFocusAfterClose={true}
      style={
        Object {
          "content": Object {
            "border": "0",
            "borderRadius": "4px",
            "bottom": "auto",
            "left": "50%",
            "maxWidth": "60rem",
            "minHeight": "10rem",
            "minWidth": "20rem",
            "opacity": "1",
            "padding": "0px",
            "position": "fixed",
            "right": "auto",
            "top": "50%",
            "transform": "translate(-50%,-50%)",
            "width": "600px",
          },
          "overlay": Object {
            "backgroundColor": "rgba(0,0,0,0.3)",
          },
        }
      }
    >
      <ModalPortal
        ariaHideApp={true}
        bodyOpenClassName="ReactModal__Body--open"
        closeTimeoutMS={0}
        defaultStyles={
          Object {
            "content": Object {
              "WebkitOverflowScrolling": "touch",
              "background": "#fff",
              "border": "1px solid #ccc",
              "borderRadius": "4px",
              "bottom": "40px",
              "left": "40px",
              "outline": "none",
              "overflow": "auto",
              "padding": "20px",
              "position": "absolute",
              "right": "40px",
              "top": "40px",
            },
            "overlay": Object {
              "backgroundColor": "rgba(255, 255, 255, 0.75)",
              "bottom": 0,
              "left": 0,
              "position": "fixed",
              "right": 0,
              "top": 0,
            },
          }
        }
        isOpen={true}
        parentSelector={[Function]}
        portalClassName="ReactModalPortal"
        shouldCloseOnEsc={true}
        shouldCloseOnOverlayClick={true}
        shouldFocusAfterRender={true}
        shouldReturnFocusAfterClose={true}
        style={
          Object {
            "content": Object {
              "border": "0",
              "borderRadius": "4px",
              "bottom": "auto",
              "left": "50%",
              "maxWidth": "60rem",
              "minHeight": "10rem",
              "minWidth": "20rem",
              "opacity": "1",
              "padding": "0px",
              "position": "fixed",
              "right": "auto",
              "top": "50%",
              "transform": "translate(-50%,-50%)",
              "width": "600px",
            },
            "overlay": Object {
              "backgroundColor": "rgba(0,0,0,0.3)",
            },
          }
        }
      >
        <div
          aria-modal="true"
          className="ReactModal__Overlay ReactModal__Overlay--after-open"
          onClick={[Function]}
          onMouseDown={[Function]}
          style={
            Object {
              "backgroundColor": "rgba(0,0,0,0.3)",
              "bottom": 0,
              "left": 0,
              "position": "fixed",
              "right": 0,
              "top": 0,
            }
          }
        >
          <div
            className="ReactModal__Content ReactModal__Content--after-open"
            onClick={[Function]}
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            onMouseUp={[Function]}
            style={
              Object {
                "WebkitOverflowScrolling": "touch",
                "background": "#fff",
                "border": "0",
                "borderRadius": "4px",
                "bottom": "auto",
                "left": "50%",
                "maxWidth": "60rem",
                "minHeight": "10rem",
                "minWidth": "20rem",
                "opacity": "1",
                "outline": "none",
                "overflow": "auto",
                "padding": "0px",
                "position": "fixed",
                "right": "auto",
                "top": "50%",
                "transform": "translate(-50%,-50%)",
                "width": "600px",
              }
            }
            tabIndex="-1"
          >
            <SimpleModalBody
              header="Update Status"
              onCloseClick={[Function]}
              onPrimaryClick={[Function]}
              onSecondaryClick={[Function]}
              primary="update"
              secondary="cancel"
              type="status"
            >
              <Component
                onCloseClick={[Function]}
              >
                <div
                  aria-labelledby="modalTitle"
                  role="dialog"
                >
                  <Component
                    title="Update Status"
                  >
                    <header
                      className="modal-inner"
                    >
                      <h1
                        className="modal-title t-gamma no-margin"
                      >
                        Update status
                      </h1>
                    </header>
                  </Component>
                  <Component>
                    <section
                      className="modal-inner"
                    >
                      <div>
                        content
                      </div>
                    </section>
                  </Component>
                  <Component>
                    <footer
                      className="modal-footer bg-dust"
                    >
                      <div
                        className="modal-button-group row"
                      >
                        <div
                          className="modal-button_item"
                        >
                          <button
                            className="button primary"
                            onClick={[Function]}
                          >
                            update
                          </button>
                        </div>
                        <div
                          className="modal-button_item"
                        >
                          <button
                            className="button no-border"
                            onClick={[Function]}
                          >
                            cancel
                          </button>
                        </div>
                      </div>
                    </footer>
                  </Component>
                  <a
                    aria-label="Close modal"
                    className="close-reveal-modal"
                    onClick={[Function]}
                  >
                    <span
                      className="ui-icon ui-medium i-primary"
                    >
                      <svg>
                        <use
                          xlinkHref="#i-close"
                        />
                      </svg>
                    </span>
                  </a>
                </div>
              </Component>
            </SimpleModalBody>
          </div>
        </div>
      </ModalPortal>
    </Modal>
  </Modal>
</SimpleModal>
`;
